Plans Next: Cleanse plans-grid-next pkg from remaining media queries and section styles

chriskmnds opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/79010

## Proposed Changes

This is part of a series of PRs for untangling the pricing grids/components from screen/media queries and any Calypso-dependent CSS context (such as sidebar width). The end goal is to (1) make the components responsive and reusable in any context, and (2) to be able to render the grids anywhere on a page (in a modal, sidebar, a half-page column, etc.). So pdgrnI-2uQ-p2 becomes relevant and this work should set the foundations for it.

In this PR:

- we follow from the work done in https://github.com/Automattic/wp-calypso/pull/86796 and https://github.com/Automattic/wp-calypso/pull/87003 to clean up remaining media queries and section styles
- migrate remaining media queries to Calypso (one left, except for the `PlanTypeSelector` which will be addressed in a follow-up)
- migrate section styles to Calypso
- remove media queries that were previously forcing the `FeaturesGrid` to snap into more condensed views at various breakpoints (see gifs below). I don't see the value in this behavior if we have enough space on the screen to fit a larger grid (respecting the max column widths). This also makes the `FeaturesGrid` behave similarly to `/pricing`, which also doesn't apply "snapping" across these breakpoints.

## Media

Features-grid Before

https://github.com/Automattic/wp-calypso/assets/1705499/ebc91dc5-cd50-4625-89b0-0e0c36bd4897

Features-grid After

https://github.com/Automattic/wp-calypso/assets/1705499/6cf0642f-54d5-46f4-aac2-3bec83fb0e7c

Features-grid on /pricing

https://github.com/Automattic/wp-calypso/assets/1705499/9b3d8787-90b0-42f7-866f-de15d17f8bc0

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- General testing across `/start/plans`, `/plans/[ any site ]`, `/setup/newsletter/plans`
- Confirm nothing broken visually on resizing (general layout, borders, highlight labels)
- Some styles were removed from `/start/hosting`. It doesn't look like they were ever applied or effective at all

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

chriskmnds committed:
css media query refactors - cleanse the rest

chriskmnds committed:
css media query refactors - cleanse the rest

chriskmnds committed:
css media query refactors - cleanse the rest

chriskmnds committed:
css media query refactors - cleanse the rest

chriskmnds committed:
css media query refactors - cleanse the rest - with breakage testing /start/hosting flow

chriskmnds committed:
css media query refactors - cleanse the rest - remove hosting flow styles

chriskmnds committed:
css media query refactors - cleanse the rest - migrate step-wrapper__header styles

chriskmnds committed:
css media query refactors - cleanse the rest - remove unused formatter-header

chriskmnds committed:
css media query refactors - cleanse the rest - remove unused

chriskmnds commented on client/my-sites/plans-features-main/style.scss:
```diff
@@ -36,10 +39,7 @@
-		@include plans-2023-break-small {
-			padding-top: 35px; // enough to cover `plans-grid__popular-badge` repositioning (top: -35px)
-		}
+		padding-top: 35px; // enough to cover `plans-grid__popular-badge` repositioning (top: -35px)
```
This needs to be addressed separately. Best not to live under `plans-features-main`. The grids really shouldn't reposition elements while not respecting their render bounds/frame, as that's what I think is happening with the badges. If possible, the extra padding can happen within the component itself and not the wrapper's. Ideally, the repositioning wouldn't be necessary if the layouts were done more robustly (I think stuff we flagged previously: https://github.com/Automattic/wp-calypso/issues/72910).

chriskmnds replied:
issue: https://github.com/Automattic/wp-calypso/issues/87332

chriskmnds commented on client/my-sites/plans-features-main/style.scss:
```diff
@@ -58,17 +58,6 @@
 
-// Content group
-.plans-features-main__group {
-	& + & {
```
Old code that there's no use of.
`& + &` targets the adjacent sibling e.g. 

```
<div class="plans-features-main__group">
<div class="plans-features-main__group">
```

chriskmnds commented on client/my-sites/plans-features-main/style.scss:
```diff
@@ -58,17 +58,6 @@
-		padding-top: 19px; // popular banner height adjustment
-	}
-
-	+ .faq {
```
Similar. We don't render any sibling with the `.faq` class.

chriskmnds commented on packages/plans-grid-next/src/style.scss:
```diff
@@ -707,42 +703,19 @@ $mobile-card-max-width: 440px;
 		margin-bottom: auto;
+	}
 
-		&-buttons {
```
This pattern makes it hard to search & debug things. You cannot reach the full class from the browser in the editor. I'd avoid like fire 😬

aneeshd16 reviewed (approved):
I tested a few screens and code changes LGTM! It's great to see the amount of "red" changes in this PR! 🎉  

I noticed a minor regression - the padding between term dropdown and the grid is less compared to production.

* /start
**This PR**
<img width="976" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/4d7acc3d-55ef-4650-b32a-1e15a10674ff">

**Production**
<img width="983" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/350861fe-2dfd-43af-a293-ec6a1c447b71">

* /plans
**This PR**
<img width="882" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/adf7ee83-f8c7-45d1-8553-1b57918f5d07">

**Production**
<img width="895" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/87ca7b3a-4dd9-437b-b049-2b78de670b05">

chriskmnds commented:
> Thanks @aneeshd16 for taking the time. Red is good - in code and stocks (not financial advice) 😁
> 
> Concerning the regression, I had removed some awfully-suspicious padding (`19px`) that contributed. re-added now. these are being cleaned up in another PR that separates out page-layout and grid concerns a little more. some difference still exists, although it ooks quite minor

Actually, I've reverted this last change as it adds the extra padding to mobile now. Let's revisit the padding in the follow-up PR (will post later), as they work a little differently there. I presume we can live with the regression for a day or two - looks pretty minor 😬
